airflow/models/dagrun.py

```diff
@@ -258,6 +281,14 @@ def validate_run_id(self, key: str, run_id: str) -> str | None:
             )
         return run_id
 
+    @hybrid_property
+    def conf(self):
+        return self._conf
+
+    @conf.setter
+    def conf(self, value):
+        self._conf = ConfDict(value)
```

Comment:
Does this need to check if the incoming value is already a ConfDict?

Reply:
@uranusjr I think it doesn't